src/TestFramework/TestFramework/Attributes/DataSource/DataRowAttribute.cs

```diff
@@ -15,25 +15,279 @@ namespace Microsoft.VisualStudio.TestTools.UnitTesting;
 [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
 public class DataRowAttribute : Attribute, ITestDataSource
 {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DataRowAttribute"/> class.
+    /// </summary>
+    public DataRowAttribute()
```

Comment:
IMO we should **not** keep this one because that's equivalent to not having any data (we expect parameterless method). WDYT?

Reply:
Are you confident enough that this will not break people?

Reply:
If they were using this case, it would break them for sure but the case makes no functional sense.

I mean if you write:

```csharp
[TestMethod]
[DataRow()]
public void A() {}
```

Why not simply doing:

```csharp
[TestMethod]
public void A() {}
```

Reply:
Any such case in github?

Reply:
Looking at GH, it seems that some people are using it (mainly us but also some others):
https://grep.app/search?q=%5BDataRow%28%29%5D

Reply:
Otherwise we keep and mark as `Obsolete` so we can remove in v4. Sounds like best course of action.